src/components/LinkApp.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+import { Link as RouterLink, LinkProps as RouterLinkProps } from 'react-router-dom';
+import styled from 'styled-components';
+import {
+  color,
+  colorStyle,
+  fontStyle,
+  layout,
+  lineHeight,
+  size,
+  space,
+  textStyle,
+  typography
+} from 'styled-system';
+
+import { TextProps } from '@components';
+import { textVariants } from '@theme';
+
+const SLink = styled.a<TextProps & HTMLAnchorElement>`
+  ${textVariants}
+  ${space}
+  ${fontStyle}
+  ${color}
+  ${size}
+  ${colorStyle}
+  ${textStyle}
+  ${lineHeight}
+  ${typography}
+  ${layout}
+  ${({ textTransform }) => textTransform && { 'text-transform': textTransform }}
+`;
+
+const SRouterLink = styled(RouterLink)<TextProps & RouterLinkProps>`
+  ${textVariants}
+  ${space}
+  ${fontStyle}
+  ${color}
+  ${size}
+  ${colorStyle}
+  ${textStyle}
+  ${lineHeight}
+  ${typography}
+  ${layout}
+  ${({ textTransform }) => textTransform && { 'text-transform': textTransform }}
+`;
+
+interface LinkProps {
+  readonly href: string;
+  readonly isExternal?: boolean;
+  readonly variant?: 'inlineLink' | 'defaultLink';
+  onClick?(): void;
+}
+
+type LinkAppProps = LinkProps & Omit<RouterLinkProps, 'to'>;
+
+const LinkApp: React.FC<LinkAppProps & TextProps> = ({
+  href,
+  isExternal = false,
+  variant = 'defaultLink',
+  onClick,
+  ...props
+}) => {
+  const handleClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    if (!onClick) return;
+    event.preventDefault();
+    onClick();
+  };
+
+  if (isExternal) {
+    return (
+      <SLink
+        href={href}
+        variant={variant}
+        target="_blank"
+        rel="noreferrer"
+        onClick={handleClick}
+        {...props}
+      />
+    );
+  } else {
+    return <SRouterLink to={href} variant={variant} onClick={handleClick} {...props} />;
+  }
+};
```

Comment:
Using a ternary can save the use a return statement and improve readability IMO

Reply:
Done